store/sqlstore/channel_store.go

```diff
@@ -1442,6 +1442,27 @@ func (s SqlChannelStore) GetChannelMembersTimezones(channelId string) store.Stor
 	})
 }
 
+func (s SqlChannelStore) GetChannelMembersEmails(channelId string) store.StoreChannel {
+	return store.Do(func(result *store.StoreResult) {
+		var members []*model.ChannelMemberForExport
+		_, err := s.GetReplica().Select(&members, `
+					SELECT
+						Users.Email as UserName
```

Comment:
UserEmail would be more appropriate field name

Reply:
I feel why we need to add all this code. Why just using User.GetAllProfilesInChannel doesn't work for us? I understand it gets unnecessary details and looks to be slower. But  it has caching as well. So it can be faster if cache is hit and we will not have to manage all this code in cherrypicks.

Reply:
ya we can use the existing functionality _c.App.GetChannelMembersPage(c.Params.ChannelId, c.Params.Page, c.Params.PerPage)_, only thing need to be added extra will be pagination to get all the members of channel

Reply:
No, I am not talking about GetChannelMembersPage. I am saying you can create a method in app/channel.go getAllUserEmailsInChannel which calls a.Srv.Store.User().GetAllProfilesInChannel(channel.Id, false) and pareses GetAllProfilesInChannel and returns only the email. You dont need any pagination.

Reply:
Now I am using the existing GetAllProfilesInChannel method, so my store method is not used anymore now.